xstate/scxml.py

```diff
@@ -1,33 +1,82 @@
-from typing import List
-from xstate.transition import Transition
-
-# defaultHistoryContent = new HashTable()
-# computeEntrySet(enabledTransitions, statesToEnter, statesForDefaultEntry, defaultHistoryContent)
-# for s in statesToEnter.toList().sort(entryOrder):
-#     configuration.add(s)
-#     statesToInvoke.add(s)
-#     if binding == "late" and s.isFirstEntry:
-#         initializeDataModel(datamodel.s,doc.s)
-#         s.isFirstEntry = false
-#     for content in s.onentry.sort(documentOrder):
-#         executeContent(content)
-#     if statesForDefaultEntry.isMember(s):
-#         executeContent(s.initial.transition)
-#     if defaultHistoryContent[s.id]:
-#         executeContent(defaultHistoryContent[s.id])
-#     if isFinalState(s):
-#         if isSCXMLElement(s.parent):
-#             running = false
-#         else:
-#             parent = s.parent
-#             grandparent = parent.parent
-#             internalQueue.enqueue(new Event("done.state." + parent.id, s.donedata))
-#             if isParallelState(grandparent):
-#                 if getChildStates(grandparent).every(isInFinalState):
-#                     internalQueue.enqueue(new Event("done.state." + grandparent.id))
-
-
-def enter_states(enabled_transitions: List[Transition]):
-    states_to_enter = set()
-    states_for_default_entry = set()
+import xml.etree.ElementTree as ET
+from typing import Optional
+import json
 
+from xstate.machine import Machine
+
+ns = {"scxml": "http://www.w3.org/2005/07/scxml"}
+
+
+def convert_scxml(element: ET.Element, parent):
+    states = element.findall("scxml:state", namespaces=ns)
+
+    return {
+        "id": "machine",
+        "initial": element.attrib.get("initial", None),
+        "states": accumulate_states(element, parent),
+    }
+
+
+def accumulate_states(element: ET.Element, parent: ET.Element):
```

Comment:
This converts a list of states to a dictionary, so instead of `[{key: 'foo', ...}, {key: 'bar', ...}]` it becomes `{ foo: ..., bar: ... }`